### PR TITLE
TUI for user configuration done

### PR DESCRIPTION
**Pull request by SaurusXI**
(Mostly done)
- Needs testing
- Needs integration once local API daemon is implemented

@L04DB4L4NC3R

### CODE REVIEW

**Comment by Geek-ubaid** [comment]
@L04DB4L4NC3R review the PR and merge it, its getting stale.

File: settings/tui/tui.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// package main
+
+// import (
+// 	"strings"
+
+// 	"github.com/gdamore/tcell"
+// 	"github.com/rivo/tview"
+// )
+
+// // 1,000 most common English words.
```
Comment (L04DB4L4NC3R) [inline settings/tui/tui.go]: Why is this being committed ?

File: settings/tui/tui.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+
+package main
+
+import (
+	"github.com/rivo/tview"
+	"settings/tui/screen"
+)
+
+func main() {
```
Comment (L04DB4L4NC3R) [inline settings/tui/tui.go:10]: Setup log level on fn main

File: settings/tui/screen/screen.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+		SetRows(1, 1, 1, 1, 1, 1, -1).
+		AddItem(fb.Field, 0, 0, 1, 1, 0, 0, true).
+		AddItem(db.Field, 1, 0, 1, 1, 0, 0, true).
+		AddItem(paddingBx, 2, 0, 3, 1, 0, 0, false).
+		AddItem(bb.Grid, 5, 0, 1, 1, 0, 0, true).
+		AddItem(paddingBx, 6, 0, -1, -1, 0, 0, false)
+
+	appGrid.SetBackgroundColor(tview.Styles.PrimitiveBackgroundColor)
+	
+	return &Screen{
```
Comment (L04DB4L4NC3R) [inline settings/tui/screen/screen.go:49]: Use better variable names than bb, db

File: settings/tui/screen/screen.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	bb := elements.NewButtonsBox()
+	db := elements.NewDomainBox()
+	fb := elements.NewFreqBox()
+
+	paddingBx := tview.NewBox()
+
+	appGrid := tview.NewGrid().
+		SetColumns(-1).
+		SetRows(1, 1, 1, 1, 1, 1, -1).
+		AddItem(fb.Field, 0, 0, 1, 1, 0, 0, true).
```
Comment (L04DB4L4NC3R) [inline settings/tui/screen/screen.go]: Either use value names, or comment on what each value is in the args

File: settings/tui/screen/elements/freq.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import (
+	// "log"
+	// "strings"
+
+	"github.com/gdamore/tcell"
+	"github.com/rivo/tview"
+)
+
+const (
+	FreqFieldText = "Number of papers per day (max 5):"
```
Comment (L04DB4L4NC3R) [inline settings/tui/screen/elements/freq.go]: Keep all these texts and prompts in a single const initialisation block

File: settings/tui/screen/elements/domain.go [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+func (d *DomainBox) GetText() string {
+	return d.Field.GetText()
+}
+
+func NewDomainBox() *DomainBox {
+	d := DomainBox {
+		tview.NewInputField(),
+	}
+
+	padding := strings.Repeat(" ", len(FreqFieldText) - len(DomainFieldText))
```
Comment (L04DB4L4NC3R) [inline settings/tui/screen/elements/domain.go:24]: Explanation needed

File: settings/tui/screen/elements/buttons.go [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+func (b *ButtonsBox) RemoveButton() error {
+	if b.totalButtons == 0 {
+		return errors.New("No domains left to remove")
+	}
+
+	idx := b.currentFocus
+	btn := b.buttons[idx]
+
+	b.buttons = append(b.buttons[:idx], b.buttons[idx+1:]...)
```
Comment (L04DB4L4NC3R) [inline settings/tui/screen/elements/buttons.go:61]: Too many slicing and array deconstruction in one line. Mention the info in a comment

File: TODO.md [hunk 8]
```diff
@@ -0,0 +1,3 @@
+- Add locally hosted API daemon for settings
```
Comment (L04DB4L4NC3R) [inline TODO.md]: Shift this section to the main README

**Review by L04DB4L4NC3R — CHANGES_REQUESTED** [review]

**Comment by L04DB4L4NC3R** [comment]
> @L04DB4L4NC3R review the PR and merge it, its getting stale.

Thanks for bringing it to my notice

**Comment by L04DB4L4NC3R** [comment]
@SaurusXI do the changes

**Comment by SaurusXI** [comment]
@L04DB4L4NC3R done, please review

**Review by L04DB4L4NC3R — APPROVED** [review]

**Pull request merged**